File: code_examples/3-Structured_Problem_Solving/19-Loops_and_Ifs/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module is the Fizz Buzz program
"""

import time

def main() -> None:
    """The main() function is the Fizz Buzz program, returns None."""

    # process & output
    for counter in range(1, 101):
        if counter % 3 == 0 and counter % 5 == 0:
            print("Fizz-Buzz")
        elif counter % 3 == 0:
            print("Fizz")
        elif counter % 5 == 0:
            print("Buzz")
        else:
            print(counter)
        time.sleep(0.25)

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/3-Structured_Problem_Solving/19-Loops_and_Ifs/Java/Main.java
/*
 * This program is the Fizz Buzz program
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

public class Main {
  /**
   * This function handles all the inputs and outputs.
   *
   * @param args
   */
  public static void main(String[] args) {
    // this function is the Fizz Buzz program

    // process & output
    for (int counter = 1; counter <= 100; counter++) {
      if (counter % 3 == 0 && counter % 5 == 0) {
        System.out.println("Fizz-Buzz");
      } else if (counter % 3 == 0) {
        System.out.println("Fizz");
      } else if (counter % 5 == 0) {
        System.out.println("Buzz");
      } else {
        System.out.println(counter);
      }
    }
    System.out.println("\nDone.");
  }
}
